.github/workflows/deploy-urban.yml

```diff
@@ -0,0 +1,159 @@
+name: Deploy Urban
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main, production ]
+    paths:
+      - urban-dashboard/**
+
+jobs:
+  define-environment:
+    name: Set environment
+    runs-on: ubuntu-latest
+    outputs:
+      env_name: ${{ steps.define_environment.outputs.env_name }}
+    steps:
+      - name: Set the environment based on the branch
+        id: define_environment
+        run: |
+          if [ "${{ github.ref }}" = "refs/heads/main" ]; then
+            echo "env_name=staging" >> $GITHUB_OUTPUT
+          elif [ "${{ github.ref }}" = "refs/heads/production" ]; then
+            echo "env_name=production" >> $GITHUB_OUTPUT
+          fi
+      - name: Print the environment
+        run: echo "The environment is ${{ steps.define_environment.outputs.env_name }}"
+
+  build:
+    runs-on: ubuntu-latest
+    needs: define-environment
+    environment: ${{ needs.define-environment.outputs.env_name }}
+
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: 'urban-dashboard/'
+          sparse-checkout-cone-mode: true
+
+      - name: Read Node.js version from .nvmrc
+        id: nvm
+        run: echo "::set-output name=NODE_VERSION::$(cat urban-dashboard/.nvmrc)"
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ steps.nvm.outputs.NODE_VERSION }}
+
+      - name: Install Yarn
+        run: npm install -g yarn
+
+      - name: Build for production 🔧
+        working-directory: ./urban-dashboard
+        run: |
+          FEATURES_API_URL=$(grep '^FEATURES_API_URL=' ../.env | cut -d '=' -f 2)
+          echo REACT_APP_MAPBOX_TOKEN="${{ secrets.REACT_APP_MAPBOX_ACCESS_TOKEN }}" >> .env
+          echo REACT_APP_MAPBOX_STYLE_URL="${{ secrets.REACT_APP_MAPBOX_STYLE_URL }}" >> .env
+          echo REACT_APP_FEATURES_API_URL="${FEATURES_API_URL}" >> .env
+          echo "REACT_APP_PUBLIC_URL=$(echo "${{ vars.PUBLIC_URL_URBAN }}" | sed -E 's|^https?://[^/]+||; s|/*$|/|')" >> .env
+          echo "Checking the REACT_APP_FEATURES_API_URL value in the .env file:"
+          grep 'REACT_APP_FEATURES_API_URL' .env
```

Comment:
I think we can remove this grep from here since we don't seem to be piping it down anywhere.